examples/tests/stress.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import cpu
+from avocado.utils import memory
+from avocado.utils import process
+
+
+class Stress(Test):
+
+    """
+    Calls stress, a simple program which aims to impose certain types of
+    computing stress on the target machine.
+    @author: Yi Yang (yang.y.yi@gmail.com)
+    """
+
+    def setUp(self):
+        """
+        Build 'stress'.
+        Source:
+         http://people.seas.harvard.edu/~apw/stress/stress-1.0.4.tar.gz
+        """
+        stress_tarball = self.params.get('stress_tarball',
+                                         default='stress-1.0.4.tar.gz')
+        tarball_path = self.get_data_path(stress_tarball)
+        archive.extract(tarball_path, self.srcdir)
+        stress_version = stress_tarball.split('.tar.')[0]
+        self.srcdir = os.path.join(self.srcdir, stress_version)
+        os.chdir(self.srcdir)
+        process.run('./configure')
+        build.make(self.srcdir)
+
+    @staticmethod
+    def _freespace(path):
```

Comment:
This looks like a good candidate for a new utility function (provided it does not exist yet "elsewhere").